### PR TITLE
Upper Header Section | CID-5

### PR DESCRIPTION
**Pull request by WallochMatt**
Created an Upper Header Component and changed props/places of instantiation to utilize the new component.

### CODE REVIEW

**Comment by ChrisHaliga** [comment]
Just getting started looking at this. Would you mind including some screenshots of the changed and new components and also adding the Jira ticked id to the title?

File: Frontend/src/ContextMenu.svelte [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+
+        // console.log("x: ", position.x );
+        // console.log("y: ", position.y, );
+
+        // if (browser.w -  position.x < menu.w) {}
+        //     position.x = position.x - menu.w
+        // }
+        // if (browser.h -  position.y < menu.h) {
+        //     position.y = position.y - menu.h
```
Comment (ChrisHaliga) [inline Frontend/src/ContextMenu.svelte:33]: reminder to remove commented out code

File: Frontend/src/ContextMenu.svelte [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+
+
+    }
+
+        console.log("w: ", menu.w );
+        console.log("h: ", menu.h, );
```
Comment (ChrisHaliga) [inline Frontend/src/ContextMenu.svelte:40]: Also debugging console logs

File: Frontend/src/ContextMenu.svelte [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+
+    function offMenuClick(e){
+        showMenu  = false;
+    }
```
Comment (ChrisHaliga) [inline Frontend/src/ContextMenu.svelte:44]: Nothing wrong with this at all, I just want to make sure you know it's okay to shorten stuff like this to reduce bloat if you like.

If offMenuClick is only used one time, you can write it like: `on:click={() => showMenu  = false}`

and if it's used multiple times you could write this in lambda notation: `let offMenuClick = () => showMenu  = false;`

File: Frontend/src/ContextMenu.svelte [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+    function editItem(){
+        content.textContent = "Printed..."
+    }
+    function removeItem(){
+        content.textContent = "Removed..."
+    }
+    function setting(){
+        content.textContent = "Settings..."
+    }
```
Comment (ChrisHaliga) [inline Frontend/src/ContextMenu.svelte:66]: see my comment above on how to shorten these

File: Frontend/src/ContextMenu.svelte [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+        border: none;
+        border-bottom: 1px solid #ccc;
+        margin: 5px 0px;
+    }
+</style>
+
+
+
+{#if showMenu}
+<nav use:getContextMenuDimension style="position: absolute; top:{position.y}px; left:{position.x}px">
```
Comment (ChrisHaliga) [inline Frontend/src/ContextMenu.svelte:156]: Great use of inline styles! As I'm sure you know, in general they are bad, but this is quite clever. May want to move the non-dynamic styles (just `position:absolute` in this case) into the styles section

File: Frontend/src/Modal.svelte [hunk 6]
```diff
@@ -1,29 +1,78 @@
...
+				<button on:click={() => (showModal = false)}> X </button>
+			</div>
+			<div>
+				<h2>
+					<!-- The slice is to remove the S on a given entity. Seek better solution, 
+						it's especially awkward for "Create New Locations & Access Point" -->
+					Create New {entity.slice(0, -1)}
+					<hr />
+				</h2>
+				<slot name="form" />
```
Comment (ChrisHaliga) [inline Frontend/src/Modal.svelte:30]: I'm pretty sure this is your only slot on this component, so feel free to just write `<slot/>`

File: Frontend/src/Modal.svelte [hunk 7]
```diff
@@ -1,29 +1,78 @@
...
+	</dialog>
+{/if}
 
-		<slot name="header" />
-			<hr />
-		<slot />
+<style>
+	dialog {
+		z-index: 10;
+		/* No display: none */
```
Comment (ChrisHaliga) [inline Frontend/src/Modal.svelte:39]: remove commented code

File: Frontend/src/lib/forms/AddLocationForm.svelte [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+    <label>
+        Address
+        <input type="text" placeholder="" bind:value={address}>
+    </label>
+    <label>
+        Associated Group
+        <input type="text" placeholder="" bind:value={associatedGroup}>
+    </label>
+    <label>
+        Access Poitns
```
Comment (ChrisHaliga) [inline Frontend/src/lib/forms/AddLocationForm.svelte:26]: *Points

**Comment by ChrisHaliga** [comment]
Overall looks great! There's a lot of placeholder data, console logs, alerts, and comment reminders, so just be careful not to let all of that get ahead of you, but I really have no concerns. Just post those image references so we can verify the UI and address the little things and you should be golden

**Comment by WallochMatt** [comment]
CID-upper-header

The component had no visual changes shown. The changes are functional/organizational. That said, minor padding was added to have space between elements.

**Pull request merged**